File: CommandTalk.cpp
//class Command/Talk.cpp

#include "CommandTalk.h"
#include <iostream>
#include <string>
#include <vector>
#include "Character.h"
#include "Sentence.h"

CommandTalk::CommandTalk()
{
}


void CommandTalk::run()
{
    if (_args.size() == 0) {
        throw "Who must I talk to ?";
    }
    else if (_args.size() == 1) {
        throw "What must I say ?";
    }

    std::string characterName = _args[0];
    std::string triggerWord = _args[1];

    Character* character = Character::searchByNameAndPlayer(characterName, _player);

    std::vector <Sentence*> s = Sentence::loadByCharacterIdAndTriggerWord(character->getId(), triggerWord);

    std::cout << s.size() << std::endl;
}

File: CommandTalk.py
# -*- coding: utf8 -*-

from CommandAbstract import CommandAbstract
from CommandException import CommandException
from CharacterException import CharacterException
from Character import Character
from sentence import sentence
import random
import string


class CommandTalk(CommandAbstract):
	def run(self):
		if len(self._args) == 0:
			raise CommandException("Who must I talk to ?")
		elif len(self._args) == 1:
			raise CommandException("What must I say ?")

		characterName = self._args[0]
		triggerWord = self._args[1]
		character = Character.searchByNameAndPlayer(
			characterName, self._player
		)

		if character is None:
			raise CharacterException("Unknown Character")
		s = sentence.loadByCharacterIdAndTriggerWord(
			character.getId(), triggerWord
		)

		if len(s) is 0:
			print("What ?")
			return

		s = s[random.randint(0, len(s) - 1)]
		print(self.processSentence(
			s.getSentence(), self._player._model.getName()
		))

	def processSentence(self, s, characterName):
		return s % {'player_name': characterName}
